shared/adapters/chain/cosmos/keys.ts

```diff
@@ -1,25 +1,31 @@
 // Cosmos cannot sign arbitrary blobs, but they can sign transactions. So, as a hack around that,
 // we insert our account registration token into a proposal message, and then verify against the
 // generated signature. But first we need the message to insert.
-export const VALIDATION_CHAIN_DATA = {
-  sequence: '0', accountNumber: '0', chainId: 'validation'
-};
 
-export const keyToMsgSend = async (address: string, token: string) => {
+export const validationTokenToSignDoc = async (address: string, token: string) => {
   const CosmosApi = await import('@lunie/cosmos-api');
   const cosmos = new CosmosApi.default('');
   const jsonTx = (cosmos as any).MsgSubmitProposal(address, {
     proposalType: 'cosmos-sdk/TextProposal',
     title: token.trim(),
     description: '',
-    initialDeposits: [{ denom: 'stake', amount: '0'}]
+    initialDeposits: [{ denom: 'stake', amount: '0' }]
   }).message;
-  const stdTx = CosmosApi.createStdTx({ gas: '0', gasPrices: [{ denom: 'stake', amount: '0' }], memo: ''}, jsonTx);
-  return stdTx;
-};
+  const stdTx = CosmosApi.createStdTx({
+    gas: '100000',
+    gasPrices: [{ denom: 'stake', amount: '0' }],
+    memo: ''
+  }, jsonTx);
+  delete stdTx.fee;
```

Comment:
oof

Reply:
If you look inside CosmJS vs Lunie vs gaiacli vs the two or three other libraries, this kind of thing actually happens in quite a few places - downside of having too many enthusiastic implementers...

Also, it's been a year and a half and that ICS for signing text is still open